src/main/java/hudson/plugins/favorite/user/FavoriteUserProperty.java

```diff
@@ -81,16 +102,36 @@ public boolean toggleFavorite(String job) throws IOException {
         }
     }
 
-    void deleteFavourite(String job) throws IOException {
+    void deleteFavourite(@NonNull String job) throws IOException {
         data.remove(job);
         user.save();
     }
 
     @Override
-    public UserProperty reconfigure(StaplerRequest req, JSONObject form) throws Descriptor.FormException {
+    public UserProperty reconfigure(StaplerRequest req, JSONObject form) {
+        if (form != null) {
+            for (Object key : form.keySet()) {
+                String job = key.toString();
+                if (!form.getBoolean(job)) {
+                    data.remove(job);
```

Comment:
This changes the behaviour of the plugin. Previously a click on the star immediately toggled the state. Now you have to explicitly save your properties.
Also the old behaviour was doing here `data.put(job, false)` and not remove the entry from the map.

Reply:
Reverted to original implementation